rust/src/download_compile.rs

```diff
@@ -0,0 +1,133 @@
+use crate::bs_safe::{BridgeStanError, Result};
+use flate2::read::GzDecoder;
+use path_absolutize::Absolutize;
+use std::{
+    env::temp_dir,
+    fs,
+    path::{Path, PathBuf},
+};
+use tar::Archive;
+
+pub(crate) const VERSION: &str = env!("CARGO_PKG_VERSION");
+
+/// Download and unzip the BridgeStan source distribution for this version
+/// to ~/.bridgestan/bridgestan-version
+pub fn get_bridgestan_src() -> Result<PathBuf> {
+    let homedir = dirs::home_dir().unwrap_or(temp_dir());
+
+    let bs_path_download_temp = homedir.join(".bridgestan_tmp_dir");
+    let bs_path_download = homedir.join(".bridgestan");
+
+    let bs_path_download_temp_join_version =
+        bs_path_download_temp.join(format!("bridgestan-{VERSION}"));
+    let bs_path_download_join_version = bs_path_download.join(format!("bridgestan-{VERSION}"));
+
+    if !bs_path_download_join_version.exists() {
+        println!("Downloading BridgeStan");
```

Comment:
I know we do it in the other languages, but for some reason printing here feels less appropriate than in the scripting languages. OTOH, we do want to let people know what we're doing.

Reply:
I think you're right, I changed the code to use log::info instead of println